### ISSUE TITLE
Child blocks are mixed up on save

### ISSUE THREAD

**Issue by Sjouwke** [spicywebau/craft-neo#745]
### Bug Description

Child blocks are mixed up on save. For example:
- Top level block Split can only have 1 child of type Image or Video
- Top level block Cards can only have child blocks of type Card
- Top level block Testimonials can only have child blocks of Testimonials.
Suddenly, sub blocks are mixed up. Split block now has 3 children of different types. Testimonials are added to cards. Cards bock only has 1 child, of type testimonial.
This happens on multiple pages.

### Steps to reproduce

1. Save page
2. Open page again
3. Sub blocks are mixed up

### Neo version

3.7.8

### Craft CMS version

4.4.10.1

### What is the affected Neo field's propagation method?

Save blocks to all sites the owner element is saved in

### Does this issue involve templating, and if so, is eager-loading used?

This is not a templating issue

**Comment by ttempleton** [comment]
If possible, could you please send a database backup and your composer.json/lock files to plugins@spicyweb.com.au, along with an example of an entry that has this issue, and we'll have a look.

**Comment by ttempleton** [comment]
Thanks for sending those @Sjouwke. Unfortunately I haven't been able to reproduce the behaviour yet, but I've seen evidence of it having happened in the entry revisions. At this point I think it might be caused by the issue described in https://github.com/spicywebau/craft-neo/issues/708#issuecomment-1420312363 / #717 where old block structure data from provisional drafts wasn't being cleared out. Newer Neo releases (>= 3.6.3) clear that data out as part of Craft's garbage collection if any still exists - could you please run `php craft gc/run` and let me know if the issue still occurs?

**Comment by Tim-Wils** [comment]
@ttempleton @Sjouwke Maybe related? https://github.com/craftcms/cms/issues/13256

**Comment by ttempleton** [comment]
@Tim-Wils There have been long-term issues with Preparse Field and Neo compatibility (see https://github.com/besteadfast/craft-preparse-field/issues/30, #398 and many other issues linked to that one) to the point that [we listed Preparse Field in the plugin incompatibility list for the Craft 3 version of Neo](https://github.com/spicywebau/craft-neo/blob/aaa3f138664afa9291db0f9fbd76aa1fe9a75157/docs/plugin-compatibility.md). I'd removed it from that list for the Craft 4 version, and hadn't heard of any issues between the two plugins on Craft 4 until now - but if there are still issues on the Preparse Field end, I'll need to add it back to the list. (Of course, if any issues turn out to be caused by Neo, we'll look into a fix for them.)

**Comment by ttempleton** [comment]
This turned out to be related to merging of live entry content into draft content, where newly-created blocks from the live entry could get merged into the wrong place in the draft block structure. The fix for that will be included in the next release, which will hopefully be after #746 is resolved. In the meantime, you can get that fix by setting your composer.json file's `spicyweb/craft-neo` requirement to `3.x-dev as 3.7.9` and running `composer update`.
